storage/local/interface.go

```diff
@@ -26,6 +26,8 @@ import (
 // are goroutine-safe. Storage implements storage.SampleAppender.
 type Storage interface {
 	prometheus.Collector
+	Querier
```

Comment:
Well, that's the mistake right there ;)
Why should a constructor for X ever not return X? Why downgrade yourself from the beginning?

That said, I did that plenty of time myself in the past but found it to always become painful at some point.